### PR TITLE
Add external transport API

### PR DESCRIPTION
**Pull request by pablogs9**
TODO:
- [x] Review in detail @jamoralp

### CODE REVIEW

**Comment by pablogs9** [comment]
This PR should target `foxy`

File: rmw_microxrcedds_c/include/rmw_uros/options.h [hunk 1]
```diff
@@ -17,12 +17,40 @@
...
 
+#ifdef RMW_UXRCE_TRANSPORT_IPV4
+  #define MAX_IP_LEN 16
+#elif defined(RMW_UXRCE_TRANSPORT_IPV6)
+  #define MAX_IP_LEN 39
+#endif
+#define MAX_PORT_LEN 5
+#define MAX_SERIAL_DEVICE 50
+
+typedef struct rmw_uxrce_transport_params_t
```
Comment (jamoralp) [inline rmw_microxrcedds_c/include/rmw_uros/options.h:36]: for now, this struct declaration fits ok here, but after I finish refactoring transports init/fini functions in a separate file (namely, transports.h) I could move it there. What do you think @pablogs9 ¿

**Review by jamoralp — APPROVED** [review]

File: rmw_microxrcedds_c/include/rmw_uros/options.h (hunk 1 above)
Comment (pablogs9) [reply rmw_microxrcedds_c/include/rmw_uros/options.h:36]: Ok

File: rmw_microxrcedds_c/src/rmw_init.c [hunk 2]
```diff
@@ -198,7 +208,7 @@ rmw_init(const rmw_init_options_t * options, rmw_context_t * context)
   // Micro-XRCE-DDS Client initialization
 
 #ifdef RMW_UXRCE_TRANSPORT_SERIAL
-  int fd = open(context->impl->connection_params.serial_device, O_RDWR | O_NOCTTY);
+  int fd = open(context->impl->transport_params.serial_device, O_RDWR | O_NOCTTY);
```
Comment (jamoralp) [inline rmw_microxrcedds_c/src/rmw_init.c:211]: This cannot be working. It should be changed to 
```suggestion
  int fd = open(options->impl->transport_params.serial_device, O_RDWR | O_NOCTTY);
```

**Review by jamoralp — CHANGES_REQUESTED** [review]

**Review by jamoralp — APPROVED** [review]

**Pull request merged**